code/controllers/subsystems/cleanup_ch.dm

```diff
@@ -0,0 +1,165 @@
+#define FIRE_PRIORITY_CLEANUP 20
+#define MAX_CLEANUP_QUEUE_TIME 120 //Max time in queue before we give up.
+#define URGENCY_PANIC_MODE 0.75 //Max urgency before we start to skip checks
+#define SMOOTHSTEP(variable) (max(min(6*(variable**5) - 15*(variable**4) + 10*(variable**3),1),0))
+//This is the cleanup subsystem. Its responsibility is to clean up unneeded atoms such as empty bullet casings and dead mobs in an effort to boost performance.
+SUBSYSTEM_DEF(cleanup)
+	name = "Cleanup"
+	priority = FIRE_PRIORITY_CLEANUP
+	flags = SS_BACKGROUND|SS_NO_INIT
+	wait = 1200 //Fires every 2 minutes.
+
+	var/stat_tag = "Cl" //Used for logging
+	var/urgency //How desperate are we to delete stuff?
+	var/points //How much complexity is in our queue?
+	var/list/processing = list()
+	var/list/currentrun = list()
+
+	var/datum/cleanupentry/current_thing
+
+/datum/controller/subsystem/cleanup/Recover()
+	log_debug("[name] subsystem Recover().")
+	if(SScleanup.current_thing)
+		log_debug("current_thing was: (\ref[SScleanup.current_thing])[SScleanup.current_thing]([SScleanup.current_thing.type]) - currentrun: [SScleanup.currentrun.len] vs total: [SScleanup.processing.len]")
+	var/list/old_processing = SScleanup.processing.Copy()
+	for(var/datum/D in old_processing)
+		if(istype(D))
+			processing |= D
+
+/datum/controller/subsystem/cleanup/stat_entry()
+	..("[stat_tag]:[processing.len] U:[urgency] P:[points]")
+
+/datum/controller/subsystem/cleanup/proc/get_points()
+	var/temp_points = processing.len
+	for(var/mob/mob_entry in processing)
+		if(istype(mob_entry))
+			temp_points += 2 //Mobs are worth 3x as much as atoms, since they're counted once in the first definition for temp_points, and twice here.
+	for(var/mob/living/carbon/carbon_entry in processing)
+		if(istype(carbon_entry))
+			temp_points += 2 //Carbons are worth 5x as much as atoms, since they're counted once in the first definition for temp_points, twice for being a mob, and twice for carbon.
+	return temp_points
```

Comment:
These istypes don't make any sense, because the for-loop is a typed for-loop, which does istype already. The istype() inside the loop on the first line will always be true, so is pointless.

You could avoid the istype() over all these items entirely if you kept the lists separate, then you could just check the len. Like if you had a proc to add to 'any' list, the subsystem could handle which list it should go into (or the thing itself can, I dunno).

Reply:
I guess this is mostly just habit from dealing with this codebase and wanting having redundancy checks everywhere to make sure shit doesn't randomly break everything. I'll get rid of these.